src/components/activity/ActivityRegistrator.tsx

```diff
@@ -0,0 +1,188 @@
+import { View, KeyboardAvoidingView, TouchableWithoutFeedback, Keyboard } from 'react-native';
+import { Text, IconButton, Avatar, useTheme } from 'react-native-paper';
+
+import Slider from '@react-native-community/slider';
+
+import { DatePicker } from '../DatePicker';
+import { TimePicker, getCurrentTimeRounded } from '../TimePicker';
+import { SuggestiveTextInput } from '../SuggestiveTextInput';
+
+import { useTranslation } from 'language/LanguageProvider';
+import Storage from 'storage';
+
+export const ActivityRegistrator = ({ route, navigation }: any) => {
+  // route.params contains information from activity screen
+
+  const lang = useTranslation();
+  const { iconSizes, colors } = useTheme();
+
+  const steps = 1;
+  const defaultChoise = {
+    value: lang.activityRegistratorActivityDefaultChoise,
+    isDefault: true
+  }
+  let choises = [ defaultChoise ];
+
+
+  const [values, modifyValues] = Storage.useValues();
+  const [activities, modifyActivities] = Storage.useActivities();
+  
+  const [fromTime, setFromTime] = React.useState(getCurrentTimeRounded(0, steps));
+  const [toTime, setToTime] = React.useState(getCurrentTimeRounded(1, steps));
+  
+  const [date, setDate] = React.useState(new Date());
+  
+  const [choise, setChoise] = React.useState(defaultChoise);
+  const [activityText, setActivityText] = React.useState('');
+
+  const [importance, setImportance] = React.useState(5); 
+  const [enjoyment, setEnjoyment] = React.useState(5);
+
+
+
+  // Find topics to choose from depending on input icon
+  const addTopicEntries = (topics: any) => {
+    for(let topicIndex = 0; topicIndex < topics.length; ++topicIndex) {
+      for(let entryIndex = 0; entryIndex < topics[topicIndex].entries.length; ++entryIndex){
+        const entry = topics[topicIndex].entries[entryIndex];
+
+        // If icons match, include it in choises
+        if(entry.icon == route.params.icon) {
+          choises.push({
+            value: entry.text,
+            isDefault: false
+          });
+        }
+      }
+    }
+  };
+
+  // Go through all categories in values
+  addTopicEntries(values.responsibilities);
+  addTopicEntries(values.relations);
+  addTopicEntries(values.enjoyment);
+  addTopicEntries(values.health);
+  addTopicEntries(values.work);
+
+
+  const onConfirm = () => {
+    // Check if custom text has been entered
+    const entryText = choise.isDefault ? activityText : choise.value;
+    
+    // Create entry from information entered by user
+    const entry = {
+      text: entryText,
+      icon: route.params.icon,
+      person: '', // TODO: link to value based on choise
+      meaningful: importance,
+      entertaining: enjoyment,
+    };
+
+    // If toTime returns 0 hours its the next day, count it as 24.
+    let toHour = toTime.getHours();
+    if (toHour === 0) {
+      toHour = 24;
+    }
+
+    // Splitting at 'T' to extract only the date from the ISO string.
+    const isoDateString = date.toISOString().split('T')[0];
```

Comment:
I think toISOString has timezone +0, which is not always the case for everyone. It needs to be dependent on the current locale.